x/outline-connectivity/main.go

```diff
@@ -149,14 +134,14 @@ func main() {
 				var testDuration time.Duration
 				switch proto {
 				case "tcp":
-					dialer, err := makeStreamDialer(proxyAddress, config.CryptoKey, config.Prefix)
+					dialer, err := outline.NewOutlineStreamDialer(config)
```

Comment:
We need to minimize the dependency on configs. In particular, having a config in addition to the access key is a pain.

We can have outline.NewOutlineStreamDialer(accessKey) instead, to bypass the need for a config. We are already committed to supporting accessKeys as a published format. In general, no one should be using the config in code, and doing so here sets a bad example.

This particular app, outline-connectivity, is special though, since it tries different combinations and needs to know how to build the Stream/Packet dialers. So it can't use access keys, unless it builds an access key, which is problematic.

We should never build access keys or configs to create the transports because we can create the transports directly. That's what the original code was doing. 

We can keep the outline.ParseAccessKey call, but I'd rather revert back to makeStreamDialer/makePacketDialer. For other apps, we should provide functions that take the accessKey instead.

I may be open to have functions that take the config JSON we actually support, but this app doesn't make use of that config, and access keys are easier because it's what the manager returns.

Reply:
The only reason `config`'s fields are public is because this `outline-connectivity` is using them, and it also needs to loop through all IP addresses as well.

Another way to completely hide the `config` is to introduce `ResolveOutlineStreamDialers(key string) []StreamDialer` and `ResolveOutlinePacketListeners(key string) []PacketListener`, they will loop through all IP addresses from a hostname and return a slice of `StreamDialer`s and `PacketListener`s. This interface is also applicable to multi-server strategy / online config as well. Any thoughts about this?